blocks/library/gallery/gallery-image.js

```diff
@@ -2,7 +2,7 @@
 export default function GalleryImage( props ) {
 	return (
 		<figure className="blocks-gallery-image">
-			<img src={ props.img.url } alt={ props.img.alt } />
+			<img src={ props.img.url } alt={ props.img.alt } data-postId={ props.img.id } />
```

Comment:
Going to keep the attribute inline, since it adds complexity to include data in comment since would require duplicating the data set and then parsing out isn't